source/cipher.c

```diff
@@ -88,6 +88,32 @@ static enum aws_cryptosdk_sha_version aws_cryptosdk_which_sha(enum aws_cryptosdk
     }
 }
 
+static bool alg_properties_equal(
```

Comment:
Alg properties are equivalent only if their pointers are equal. We do not construct them dynamically.